cmds/shell/cmd.go

```diff
@@ -83,7 +84,11 @@ func Execute(cmd *cobra.Command, args []string) error {
 
 	switch redirectURL.Query().Get("v") {
 	case "1":
-		return errors.New("the shell client doens't yet support v1 shells")
+		sockURL, _ = url.Parse(redirectURL.Query().Get("socketUrl"))
+		shell, err = v1client.Dial(sockURL.String(), []string{"bash"}, tty)
+		if err != nil {
+			return fmt.Errorf("could not create the shell client: %v", err)
+		}
 	case "2":
 		sockURL, _ = url.Parse(redirectURL.Query().Get("socketUrl"))
 		shell, err = v2client.Dial(sockURL.String(), []string{"bash"}, tty)
```

Comment:
The v2 protocol is defined to start the default system shell, if no command is given...
So if instead of giving `[]string{"bash"}` we pass `nil` or []string{}` (I think either would work), we should get the default system shell that the machine is configured to provide... hence, `bash` or `sh` on linux and `cmd.exe` on windows...

Also we should probably tweak this command to take command arguments, instead of:
```
taskcluster shell <taskId>
```
We do something like:
```
taskcluster shell <taskId> [--] [command...]
```
So that all these examples would work:
```sh
# Various ways to get an interactive, the first one always works
# because it gets system shell... the other work depending on what is installed
# on the remote system.
taskcluster shell $TASKID                           # Get default system shell
taskcluster shell $TASKID -- bash                   # Force starting bash from PATH
taskcluster shell $TASKID -- bash --login           # provide argument
taskcluster shell $TASKID -- sh                     # Try sh from PATH (on remote system)
taskcluster shell $TASKID -- python                 # Or what about python
taskcluster shell $TASKID -- /usr/bin/env python    # we could even do it the right way
# We could also use this to run commands, to just deplay some data:
taskcluster shell $TASKID -- lsb_release --all      # Just run a command on the remote system
taskcluster shell $TASKID -- cat /var/log/syslog    # Dump system log
taskcluster shell $TASKID -- ps aux                 # Show programs running
# Or we could dump system log as it is written
taskcluster shell $TASKID -- tail -f /var/log/syslog
```


This way the `taskcluster shell` command allows for a lot more than just getting a simple shell... at the same time it's super easy to get an interactive shell :)

Note: The `--` in `taskcluster shell $TASKID -- ` means the arguments following `--` are all positional.. hence, when we do: `taskcluster shell $TASKID -- lsb_release --all` the `--all` flag won't be interpreted as a flag by `taskcluster shell` it'll be read as an argument that is part of the command array... I'm not sure how cobra does this, but `--` is a GNU convention, so it probably supports this :)

Reply:
Since v1 protocol doesn't provide the default system shell, if no command is given... I think it just doesn't work if no command is given...

I propose that if we are in protocol v1 (legacy) and no command is given we use the default command that the `shell.html` tool on taskcluster-tools uses:
https://github.com/taskcluster/taskcluster-tools/blob/edb67d523a8302313b1046448d7cbfda33c6d196/src/views/Shell/Shell.js#L7-L24

Yes, that's a single command with a script as the last argument :) hehe
We can just embed as a constant for use in protocol v1 case (when no command is given).